ghedesigner/radial_numerical_borehole.py

```diff
@@ -145,17 +130,13 @@ def fill_radial_cell(self, radial_cell, resist_p_eq, resist_f_eq, resist_tg_eq):
                 ],
                 dtype=np.double,
             )
-        cell_summation += num_fluid_cells
-
-        # TODO: verify whether errors are possible here and raise exception if needed
-        # assert cell_summation == num_fluid_cells
+        cell_summation += self.num_fluid_cells
 
         # load convection cells
-        for idx in range(cell_summation, num_conv_cells + cell_summation):
-            j = idx - cell_summation
-            inner_radius = self.r_in_convection + j * self.thickness_conv
-            center_radius = inner_radius + self.thickness_conv / 2.0
-            outer_radius = inner_radius + self.thickness_conv
+        for j, idx in enumerate(range(cell_summation, self.num_conv_cells + cell_summation)):
+            inner_radius = self.r_in_convection + j * self.thickness_conv_cell
+            center_radius = inner_radius + self.thickness_conv_cell / 2.0
+            outer_radius = inner_radius + self.thickness_conv_cell
```

Comment:
Using `enumerate` to give an index to the local thing being iterated on, instead of tracking them separately. No change in functionality. This is changed in several places.